src/index.ts

```diff
@@ -270,8 +270,8 @@ export default class SwupScrollPlugin extends Plugin {
 	 * Check whether to scroll in `visit:start` hook
 	 */
 	onVisitStart: Handler<'visit:start'> = (visit) => {
-		this.maybeResetScrollPositions(visit);
 		this.cacheScrollPositions(visit.from.url);
+		this.maybeResetScrollPositions(visit);
```

Comment:
Switched these around to cache old URL first, then reset new URL, for cases where old and new URL are identical (if e.g. swup's new `linkToSelf` option is set to `navigate`)